File: javaSprint2GladlyGiven/src/main/java/pt/gladlyGivenApi/GladlyGiven/Example/ExampleController.java
package pt.gladlyGivenApi.GladlyGiven.Example;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("api/v1/example")
public class ExampleController {
    @Autowired
    private ExampleModelService service;

    @GetMapping("/health")
    public String health() {
        return "Java Spring API is working fine!";
    }

    @GetMapping()
    public List<ExampleModel> getAllExampleModels() {
        var models = service.getAllExampleModels();

        if (models != null && models.size() > 0) {
            return models;
        }

        return null;
    }

    @PostMapping
    public ExampleModel createExampleModel(@RequestParam String name) {
        return service.createExampleModel(name);
    }
}

File: dotNetSprint2GladyGivenWebAPI/GladyGiven/Example/ExampleController.cs
using Microsoft.AspNetCore.Mvc;

namespace GladyGivenWebAPI.Example
{
    [ApiController]
    [Route("api/v1/[controller]")]
    public class ExampleController : ControllerBase
    {
        private readonly ExampleService service;

        public ExampleController(ExampleDBContext context)
        {
            service = new ExampleService(context);
        }

        [HttpGet("~/api/health")]
        public ActionResult<string> Health()
        {
            return Ok("API is working fine!");
        }

        [HttpGet]
        public async Task<ActionResult<List<ExampleModel>>> GetAllExampleModels()
        {
            List<ExampleModel> models = await service.GetAllModels();

            if (models != null && models.Count > 0)
                return Ok(models);
            
            return NoContent();
        }

        [HttpPost]
        public async Task<ActionResult<ExampleModel>> CreateExampleModel(string name)
        {
            ExampleModel model = await service.CreateExampleModel(name);

            if (model != null)
                return Ok(model);

            return NoContent();
        }
    }
}
